src/App.tsx

```diff
@@ -5,10 +5,24 @@ import Header from './Header';
 import Login from './Login';
 import React, { SetStateAction, useEffect, useState } from 'react';
 import { supabase } from './setupSupabase';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 
 function App() {
   const { mode, renderHeader } = Header();
   const [session, setSession] = useState(null);
+  const [userId, setUserId] = useState<string>('');
```

Comment:
For all calls to the db it's better to encapsulate them inside hooks for 2 reasons : 
1. it allow for maximum reusability
2. It "hides" supabase from the components, so if we wanna ditch supabase later it's gonna be easier (we never know)

Other than that nice !